lib/src/ruby/module.cc

```diff
@@ -297,12 +297,8 @@ namespace facter { namespace ruby {
         for (auto dir : paths) {
             _additional_search_paths.emplace_back(dir);
 
-            // Get the canonical directory name
-            boost::system::error_code ec;
-            path directory = canonical(_additional_search_paths.back(), ec);
-            if (ec) {
-                continue;
-            }
+            // Get the absolute directory name
```

Comment:
Any chance you can add more comments as to why absolute over canonical (in that canonical causes errors with symlinks when the targets are not the same file system type (likely due to stat)